llama-index-integrations/embeddings/llama-index-embeddings-deepinfra/llama_index/embeddings/deepinfra/base.py

```diff
@@ -0,0 +1,215 @@
+import logging
+import os
+
+import aiohttp
+import requests
+from typing import List, Optional
+
+from llama_index.core.base.embeddings.base import BaseEmbedding
+from llama_index.core.bridge.pydantic import PrivateAttr
+from llama_index.core.callbacks.base import CallbackManager
+
+logger = logging.getLogger(__name__)
+
+"""DeepInfra Inference API URL."""
+INFERENCE_URL = "https://api.deepinfra.com/v1/inference"
+"""Environment variable name of DeepInfra API token."""
+ENV_VARIABLE = "DEEPINFRA_API_TOKEN"
+"""Default model ID for DeepInfra embeddings."""
+DEFAULT_MODEL_ID = "sentence-transformers/clip-ViT-B-32"
+"""Maximum batch size for embedding requests."""
+MAX_BATCH_SIZE = 1024
+
+
+class DeepInfraEmbeddingModel(BaseEmbedding):
+    """
+    A wrapper class for accessing embedding models available via the DeepInfra API. This class allows for easy integration
+    of DeepInfra embeddings into your projects, supporting both synchronous and asynchronous retrieval of text embeddings.
+
+    Args:
+        model_id (str): Identifier for the model to be used for embeddings. Defaults to 'sentence-transformers/clip-ViT-B-32'.
+        normalize (bool): Flag to normalize embeddings post retrieval. Defaults to False.
+        api_token (str): DeepInfra API token. If not provided,
+        the token is fetched from the environment variable 'DEEPINFRA_API_TOKEN'.
+
+    Examples:
+        >>> from llama_index.embeddings.deepinfra import DeepInfraEmbeddingModel
+        >>> model = DeepInfraEmbeddingModel()
+        >>> print(model.get_text_embedding("Hello, world!"))
+        [0.1, 0.2, 0.3, ...]
+    """
+
+    """model_id can be obtained from the DeepInfra website."""
+    _model_id: str = PrivateAttr()
+    """normalize flag to normalize embeddings post retrieval."""
+    _normalize: bool = PrivateAttr()
+    """api_token should be obtained from the DeepInfra website."""
+    _api_token: str = PrivateAttr()
+    """query_prefix is used to add a prefix to queries."""
+    _query_prefix: str = PrivateAttr()
+    """text_prefix is used to add a prefix to texts."""
+    _text_prefix: str = PrivateAttr()
+    """batch_size is the batch size for embedding requests."""
+    _batch_size: int = PrivateAttr()
+
+    def __init__(
+        self,
+        model_id: str = DEFAULT_MODEL_ID,
+        normalize: bool = False,
+        api_token: str = None,
+        callback_manager: Optional[CallbackManager] = None,
+        query_prefix: str = "",
+        text_prefix: str = "",
+        batch_size: int = MAX_BATCH_SIZE,
+    ) -> None:
+        """
+        Init params.
+        """
+        self._model_id = model_id
+        self._normalize = normalize
+        self._api_token = os.getenv(ENV_VARIABLE, api_token)
```

Comment:
shouldn't this be `api_token or os.getenv(..)` ? Otherwise it ignores the api key in the constructor